cmd/antrea-agent/options.go

```diff
@@ -397,10 +396,18 @@ func (o *Options) setK8sNodeDefaultOptions() {
 	if o.config.HostProcPathPrefix == "" {
 		o.config.HostProcPathPrefix = defaultHostProcPathPrefix
 	}
-	if features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+	if !features.DefaultFeatureGate.Enabled(features.AntreaProxy) {
+		o.config.AntreaProxy.Enable = pointer.Bool(false)
+		klog.InfoS("Feature gate `AntreaProxy` is deprecated, please use option `antreaProxy.enable` to disable AntreaProxy")
+	} else if o.config.AntreaProxy.Enable == nil {
+		o.config.AntreaProxy.Enable = pointer.Bool(true)
+	}
+	if *o.config.AntreaProxy.Enable {
 		if o.config.AntreaProxy.ProxyLoadBalancerIPs == nil {
-			o.config.AntreaProxy.ProxyLoadBalancerIPs = new(bool)
-			*o.config.AntreaProxy.ProxyLoadBalancerIPs = true
+			o.config.AntreaProxy.ProxyLoadBalancerIPs = pointer.Bool(true)
+		}
+		if o.config.AntreaProxy.DefaultLoadBalancerMode == "" {
+			o.config.AntreaProxy.DefaultLoadBalancerMode = config.LoadBalancerModeNAT.String()
```

Comment:
After it's moved here, empty dsr mode in configuration file and disabling antreaProxy will make agent fail to start.

Reply:
Updated for this.